packages/libs/components/src/map/MapVEuMap.tsx

```diff
@@ -462,3 +430,139 @@ function MapScrollWheelZoom(props: MapScrollWheelZoomProps) {
 
   return null;
 }
+
+// custom zoom control
+function CustomZoomControl() {
+  const map = useMap();
+
+  // zoom in
+  const zoomIn = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    // map.doubleClickZoom.disable();
+    map.setZoom(map.getZoom() + 1);
+  };
+
+  // zoom out
+  const zoomOut = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    if (map.getZoom() > 2) map.setZoom(map.getZoom() - 1);
+  };
+
+  // zoom to whole world
+  const zoomToWholeWorld = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    // for SAM (or full screen mode), very small number does not work: 0.25, 0.25 is ok
+    map.setView([0.25, 0.25], 2);
+  };
+
+  // zoom to data: using flyTo function implicitly
+  const zoomToData = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    map.setView([0, 0], 2);
```

Comment:
This will need to be not-hardcoded...

Somehow!

Reply:
@bobular yes you are right. That's what I am thinking too. perhaps passing a prop like default viewport to the MapVEuMap as the default is pre-defined at MapViz/SAM (not sure of FSM but I guess it is?).

Reply:
> Sounds good. We really need to retire the FSM ASAP. It's causing maintenance overhead already...

I'm happy to do this in a separate PR